packages/contracts/contracts/Unirep.sol

```diff
@@ -94,17 +94,23 @@ contract Unirep is IUnirep, VerifySignature {
     function manualUserSignUp(
         uint48 epoch,
         uint256 identityCommitment,
-        uint256 stateTreeLeaf,
+        uint256 leafIdentityHash,
         uint256[] calldata initialData
     ) public {
-        _userSignUp(epoch, identityCommitment, stateTreeLeaf);
+        if (initialData.length == 0) revert ZeroInitialData();
```

Comment:
I think we should allow `0` data signups. We should use the pre-calculated 0 data hash for this.

Maybe overload this function with 

```solidity
    function manualUserSignUp(
        uint48 epoch,
        uint256 identityCommitment,
        uint256 leafIdentityHash
    ) public
```

Reply:
I prefer not to overload the function or it will be called like `unirepContract['manualUserSignUp(uint48,uint256,uint256)']`